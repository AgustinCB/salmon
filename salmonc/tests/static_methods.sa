class DevonshireCream {
  serveOn() {
    return "Scones";
  }
  class serveWith(condiment) {
    return condiment;
  }
}

var obj = DevonshireCream();
print obj.serveOn();
print DevonshireCream.serveWith("something");
print obj.serveWith("Nono");
print DevonshireCream.serveOn();
