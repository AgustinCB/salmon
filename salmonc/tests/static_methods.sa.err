Constants: [SimpleValue(Function { ip: 186, arity: 1, uplifts: None }), SimpleValue(String(0)), SimpleValue(Function { ip: 188, arity: 2, uplifts: None }), SimpleValue(String(6)), SimpleValue(String(16)), SimpleValue(String(23)), SimpleValue(String(33)), SimpleValue(String(42)), SimpleValue(Object { address: 59, tags: 171 }), SimpleValue(String(178)), SimpleValue(String(185)), SimpleValue(Integer(1)), SimpleValue(Integer(4)), SimpleValue(Integer(3)), SimpleValue(String(191)), SimpleValue(String(192)), SimpleValue(String(207)), SimpleValue(String(222)), SimpleValue(String(231)), SimpleValue(String(248)), SimpleValue(String(252))]
Instructions: [Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: SetGlobal(1), location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(2), location: 2 }, Instruction { instruction_type: SetGlobal(2), location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: Constant(9), location: 2 }, Instruction { instruction_type: Constant(10), location: 2 }, Instruction { instruction_type: Constant(8), location: 2 }, Instruction { instruction_type: AddTag, location: 2 }, Instruction { instruction_type: AddTag, location: 2 }, Instruction { instruction_type: SetGlobal(3), location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: GetGlobal(3), location: 3 }, Instruction { instruction_type: Duplicate, location: 3 }, Instruction { instruction_type: CheckType(7), location: 3 }, Instruction { instruction_type: JmpIfFalse(3), location: 3 }, Instruction { instruction_type: Constant(10), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: RemoveTag, location: 3 }, Instruction { instruction_type: Call, location: 3 }, Instruction { instruction_type: SetGlobal(0), location: 3 }, Instruction { instruction_type: Pop, location: 3 }, Instruction { instruction_type: Constant(14), location: 4 }, Instruction { instruction_type: GetGlobal(0), location: 4 }, Instruction { instruction_type: Duplicate, location: 4 }, Instruction { instruction_type: Constant(10), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: CheckTag, location: 4 }, Instruction { instruction_type: JmpIfFalse(4), location: 4 }, Instruction { instruction_type: Constant(15), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Jmp(12), location: 4 }, Instruction { instruction_type: Constant(16), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectHas, location: 4 }, Instruction { instruction_type: JmpIfFalse(5), location: 4 }, Instruction { instruction_type: Constant(16), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Call, location: 4 }, Instruction { instruction_type: Jmp(3), location: 4 }, Instruction { instruction_type: Constant(4), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Duplicate, location: 4 }, Instruction { instruction_type: CheckType(7), location: 4 }, Instruction { instruction_type: JmpIfFalse(3), location: 4 }, Instruction { instruction_type: Constant(10), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: RemoveTag, location: 4 }, Instruction { instruction_type: Call, location: 4 }, Instruction { instruction_type: ToStr, location: 4 }, Instruction { instruction_type: StringConcat, location: 4 }, Instruction { instruction_type: Push, location: 4 }, Instruction { instruction_type: Strlen, location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: Constant(11), location: 4 }, Instruction { instruction_type: Constant(13), location: 4 }, Instruction { instruction_type: Constant(12), location: 4 }, Instruction { instruction_type: Syscall, location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(14), location: 5 }, Instruction { instruction_type: Constant(17), location: 5 }, Instruction { instruction_type: GetGlobal(3), location: 5 }, Instruction { instruction_type: Duplicate, location: 5 }, Instruction { instruction_type: Constant(10), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: CheckTag, location: 5 }, Instruction { instruction_type: JmpIfFalse(4), location: 5 }, Instruction { instruction_type: Constant(7), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectGet, location: 5 }, Instruction { instruction_type: Jmp(12), location: 5 }, Instruction { instruction_type: Constant(18), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectHas, location: 5 }, Instruction { instruction_type: JmpIfFalse(5), location: 5 }, Instruction { instruction_type: Constant(18), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectGet, location: 5 }, Instruction { instruction_type: Call, location: 5 }, Instruction { instruction_type: Jmp(3), location: 5 }, Instruction { instruction_type: Constant(6), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectGet, location: 5 }, Instruction { instruction_type: Duplicate, location: 5 }, Instruction { instruction_type: CheckType(7), location: 5 }, Instruction { instruction_type: JmpIfFalse(3), location: 5 }, Instruction { instruction_type: Constant(10), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: RemoveTag, location: 5 }, Instruction { instruction_type: Call, location: 5 }, Instruction { instruction_type: ToStr, location: 5 }, Instruction { instruction_type: StringConcat, location: 5 }, Instruction { instruction_type: Push, location: 5 }, Instruction { instruction_type: Strlen, location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: Constant(11), location: 5 }, Instruction { instruction_type: Constant(13), location: 5 }, Instruction { instruction_type: Constant(12), location: 5 }, Instruction { instruction_type: Syscall, location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(14), location: 6 }, Instruction { instruction_type: Constant(19), location: 6 }, Instruction { instruction_type: GetGlobal(0), location: 6 }, Instruction { instruction_type: Duplicate, location: 6 }, Instruction { instruction_type: Constant(10), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: CheckTag, location: 6 }, Instruction { instruction_type: JmpIfFalse(4), location: 6 }, Instruction { instruction_type: Constant(7), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: ObjectGet, location: 6 }, Instruction { instruction_type: Jmp(12), location: 6 }, Instruction { instruction_type: Constant(18), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: ObjectHas, location: 6 }, Instruction { instruction_type: JmpIfFalse(5), location: 6 }, Instruction { instruction_type: Constant(18), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: ObjectGet, location: 6 }, Instruction { instruction_type: Call, location: 6 }, Instruction { instruction_type: Jmp(3), location: 6 }, Instruction { instruction_type: Constant(6), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: ObjectGet, location: 6 }, Instruction { instruction_type: Duplicate, location: 6 }, Instruction { instruction_type: CheckType(7), location: 6 }, Instruction { instruction_type: JmpIfFalse(3), location: 6 }, Instruction { instruction_type: Constant(10), location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: RemoveTag, location: 6 }, Instruction { instruction_type: Call, location: 6 }, Instruction { instruction_type: ToStr, location: 6 }, Instruction { instruction_type: StringConcat, location: 6 }, Instruction { instruction_type: Push, location: 6 }, Instruction { instruction_type: Strlen, location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: Constant(11), location: 6 }, Instruction { instruction_type: Constant(13), location: 6 }, Instruction { instruction_type: Constant(12), location: 6 }, Instruction { instruction_type: Syscall, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(14), location: 7 }, Instruction { instruction_type: GetGlobal(3), location: 7 }, Instruction { instruction_type: Duplicate, location: 7 }, Instruction { instruction_type: Constant(10), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: CheckTag, location: 7 }, Instruction { instruction_type: JmpIfFalse(4), location: 7 }, Instruction { instruction_type: Constant(15), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectGet, location: 7 }, Instruction { instruction_type: Jmp(12), location: 7 }, Instruction { instruction_type: Constant(16), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectHas, location: 7 }, Instruction { instruction_type: JmpIfFalse(5), location: 7 }, Instruction { instruction_type: Constant(16), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectGet, location: 7 }, Instruction { instruction_type: Call, location: 7 }, Instruction { instruction_type: Jmp(3), location: 7 }, Instruction { instruction_type: Constant(4), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectGet, location: 7 }, Instruction { instruction_type: Duplicate, location: 7 }, Instruction { instruction_type: CheckType(7), location: 7 }, Instruction { instruction_type: JmpIfFalse(3), location: 7 }, Instruction { instruction_type: Constant(10), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: RemoveTag, location: 7 }, Instruction { instruction_type: Call, location: 7 }, Instruction { instruction_type: ToStr, location: 7 }, Instruction { instruction_type: StringConcat, location: 7 }, Instruction { instruction_type: Push, location: 7 }, Instruction { instruction_type: Strlen, location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: Constant(11), location: 7 }, Instruction { instruction_type: Constant(13), location: 7 }, Instruction { instruction_type: Constant(12), location: 7 }, Instruction { instruction_type: Syscall, location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: Return, location: 7 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: GetLocal(1), location: 2 }, Instruction { instruction_type: Return, location: 2 }]
Locations: [Location { address: 252, line: 0 }, Location { address: 252, line: 2 }, Location { address: 252, line: 5 }, Location { address: 252, line: 9 }, Location { address: 252, line: 10 }, Location { address: 252, line: 11 }, Location { address: 252, line: 12 }, Location { address: 252, line: 13 }]
[stdin line 12] Property serveWith not in object
