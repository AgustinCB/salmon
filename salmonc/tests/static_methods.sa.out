Scones
something
0 - SimpleValue(String(248))
1 - SimpleValue(String(191))
